test/e2e_env/kubernetes/meshtimeout/meshtimeout.go

```diff
@@ -48,21 +48,30 @@ func MeshTimeout() {
 		Expect(mts).To(HaveLen(0))
 		Eventually(func(g Gomega) {
 			start := time.Now()
-			_, sterr, err := kubernetes.Cluster.Exec(namespace, clientPodName, "demo-client", "curl", "-v", "-H", "x-set-response-delay-ms: 5000", fmt.Sprintf("test-server_%s_svc_80.mesh", namespace))
+			_, err := client.CollectResponse(
+				kubernetes.Cluster, "demo-client", fmt.Sprintf("test-server_%s_svc_80.mesh", namespace),
+				client.FromKubernetesPod(namespace, "demo-client"),
+				client.WithHeader("x-set-response-delay-ms", "5000"),
+				client.WithMaxTime(10),
+			)
 			g.Expect(err).ToNot(HaveOccurred())
-			g.Expect(sterr).To(ContainSubstring("HTTP/1.1 200 OK"))
 			g.Expect(time.Since(start)).To(BeNumerically(">", time.Second*5))
-		}).WithTimeout(30 * time.Second).Should(Succeed())
+		}, "30s", "1s").Should(Succeed())
 
 		// when
 		Expect(YamlK8s(timeoutConfig)(kubernetes.Cluster)).To(Succeed())
 
 		// then
 		Eventually(func(g Gomega) {
-			stdout, _, err := kubernetes.Cluster.Exec(namespace, clientPodName, "demo-client", "curl", "-v", "-H", "x-set-response-delay-ms: 5000", fmt.Sprintf("test-server_%s_svc_80.mesh", namespace))
+			response, err := client.CollectFailure(
+				kubernetes.Cluster, "demo-client", fmt.Sprintf("test-server_%s_svc_80.mesh", namespace),
+				client.FromKubernetesPod(namespace, "demo-client"),
+				client.WithHeader("x-set-response-delay-ms", "5000"),
+				client.WithMaxTime(10), // we don't want 'curl' to return early
+			)
 			g.Expect(err).ToNot(HaveOccurred())
-			g.Expect(stdout).To(ContainSubstring("upstream request timeout"))
-		}).WithTimeout(30 * time.Second).Should(Succeed())
+			g.Expect(response.ResponseCode).To(Equal(504))
+		}, "30s", "1s").Should(Succeed())
```

Comment:
Didn't we set this as a default somewhere?

Reply:
Yes, we have, but I think explicit values are preferable. IIRC @jakubdyszkiewicz has an opinion on that